### ISSUE TITLE
File Fuzzing Searching

### ISSUE THREAD

**Issue by JosephTLyons** [marta-file-manager/marta-issues#738]
Just found this absolutely beautiful application.  Going to use it on my work computer for a few days and really give it a good test drive.  I've had some time to play with it and it really just feels like a Finder built for developers, since it has the same sort of command operations as Sublime/Atom/VS Code, with the Command+P and Command+Shift+P actions.

I think it would be great if the files could be searched for using fuzzy search, and not just on prefix or suffix.  It would be great if fuzzy searching was available:

- In the Command+P search modal
- When you are typing while in a directory to start to narrow down on a specific file

With fuzzy searching, if you have a file named "aaaa_bbbb_cccc_dddd.pdf," you can just search "abcd" and it should be narrowed down to.

Anyway, just a suggestion.  I really love what I'm seeing here.  Will likely donate.  Thanks!

**Comment by yanex** [comment]
Hello @JosephTLyons! Thank you very much for the kind words!
I'm afraid it would be hard to implement fuzzy search in the Look Up action (Cmd+P) as Spotlight indices don't understand pattern queries.
However, nothing prevents from supporting fuzzy search in Quick Search. I also tend to think it should become the default behavior.

**Comment by JosephTLyons** [comment]
> Hello @JosephTLyons! Thank you very much for the kind words!
> I'm afraid it would be hard to implement fuzzy search in the Look Up action (Cmd+P) as Spotlight indices don't understand pattern queries.
> However, nothing prevents from supporting fuzzy search in Quick Search. I also tend to think it should become the default behavior.

Yeah that is understandable with the Look Up action.  And yes, I think the fuzzy search would provide all the same functionality as the current search options on Quick Search, and it would be more flexible.  That would be great. :)

**Comment by JosephTLyons** [comment]
This could be useful

https://github.com/krisk/fuse-swift

**Comment by muescha** [comment]
i would also like to see a fuzzy search in the panes / tabs so i get faster to a matching folder/file
